client/me/purchases/purchase-item/index.jsx

```diff
@@ -137,138 +159,145 @@ class PurchaseItem extends Component {
 		return null;
 	}
 
-	placeholder() {
-		return (
-			<span className="purchase-item__wrapper">
-				<div className="purchase-item__plan-icon" />
-				<div className="purchase-item__details">
-					<div className="purchase-item__title" />
-					<div className="purchase-item__purchase-type" />
-					<div className="purchase-item__purchase-date" />
-				</div>
-			</span>
-		);
-	}
+	getPurchaseType() {
+		const { purchase, site, translate, slug, showSite } = this.props;
+
+		if ( showSite ) {
+			return translate( '%(purchaseType)s for {{button}}%(site)s{{/button}}', {
+				args: {
+					purchaseType: purchaseType( purchase ),
+					site: site.name,
```

Comment:
This is where I'm getting the fatal error/whitescreen (there could be others too, of course, but this is the first one to trigger it): `Uncaught TypeError: Cannot read property 'name' of null at PurchaseItem.getPurchaseType`

Steps to reproduce:
1. Connect a Jetpack site.
2. Purchase a subscription on that site.
3. Disconnect the Jetpack site.

So the code probably needs to handle the empty site case better.

Interestingly, I think I sometimes also saw the same error (and same whitescreen) on certain page loads of a different account (one that didn't have a disconnected Jetpack site) but I wasn't able to reliably reproduce it there.  I wonder if there's also something going on where the `site` object isn't always loaded by the time it's passed in here.

Reply:
Good find! thanks for calling this out. I think we can check for site to display this particular message and if there isn't one, we can just show the product name.

Reply:
I've updated the component so that the disconnected messaging shows in the status column:

**Before**
![image](https://user-images.githubusercontent.com/6981253/100376628-dc959300-2fdd-11eb-81f2-4f03faf24328.png)


**After**
![image](https://user-images.githubusercontent.com/6981253/100376522-aa843100-2fdd-11eb-8172-25d0030dd779.png)

The changes are in this commit: https://github.com/Automattic/wp-calypso/pull/47610/commits/dc2885fded8f4110d3cd4fb0818c90aacf961a8f

Reply:
I also just added a custom icon for disconnected sites:

![image](https://user-images.githubusercontent.com/6981253/100387480-2ee2ae00-2ff6-11eb-8e7f-c5cab4455b2d.png)